.github/workflows/main_event.yml

```
@@ -1,131 +1,107 @@
+      - uses: actions/checkout@v4
+
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v5.3.0
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install ruff
+      - name: Analysing the code with ruff
+        run: |
```

actionlint

shellcheck reported issue in this script: SC2046:warning:1:36: Quote this to prevent word splitting

(shellcheck)